activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -226,7 +238,17 @@ def seconds_idle # :nodoc:
       end
 
       def unprepared_statement
-        @prepared_statement_status.bind(false) { yield }
+        hash = nil
+
+        if @prepared_statements_default == false
```

Comment:
Why not we go like this?
```suggestion
        if @prepared_statements_default
          hash = Thread.current[:ar_prepared_statements] ||= {}
          hash[self.object_id] = false
        end
```

Reply:
Thanks, cleaned this up!